src/macros.rs

```diff
@@ -6,7 +6,10 @@
 /// at most once in the whole lifetime of the program.
 ///
 /// # Note
-/// this macro is unsound on multi-core systems
+///
+/// this macro requires a `critical-section` implementation to be set. For single core systems, you can
```

Comment:
Maybe feature-gate this macro with `critical-section-single-core`?

Reply:
The macro can still be used if `critical-section-single-core` is not enabled, if the user supplies a c-s implementation from somewhere else (such as a chip-specific impl for a multicore chip, from its HAL crate).